data/core/macros/weapon_specials.cfg

```diff
@@ -254,3 +254,395 @@
+        special_note=_ "Delivers a full-power, uncountered attack when enemies move adjacent. Activates once per turn."
+        [filter_self] #so this shows as inactive in combat preview, since the ability doesn't affect regular combat at all
+            [not]
+            [/not]
+        [/filter_self]
+        uses_per_turn = 1 #sets how often per turn the weapon special can be triggered
+        #strike_limit = 1 #sets maximum amount of strikes, still bounded by weapons strike number
+        {WEAPON_SPECIAL_INTERCEPT_EVENT}
+    [/extra_attack]
+#enddef
+
+#define WEAPON_SPECIAL_INTERCEPT_EVENT
+    [event]
+        name=side turn
+        first_time_only=no
+        id=weapon_special_intercept_recharge_1
+        [lua]
+            code=<<
+local units = wesnoth.units.find_on_map({formula="self.wml_vars.intercept_used"})
+for _,u in ipairs(units) do
+    u.variables.intercept_used = 0
+end
+>>
+        [/lua]
+    [/event]
+    [event]
+        name=recruit,recall
+        id=weapon_special_intercept_recharge_2
+        first_time_only=no
+        [filter]
+            formula="self.wml_vars.intercept_used"
+        [/filter]
+        [lua]
+            code=<<
+local u = wesnoth.units.get(wesnoth.current.event_context.x1, wesnoth.current.event_context.y1)
+u.variables.intercept_used = 0
+>>
+        [/lua]
+    [/event]
+    [event]
+        name = enter_hex
+        id = intercept_weapon_special_event
+        first_time_only = no
+        [filter]
+            [filter_adjacent]
+                [has_attack]
+                    special_id = intercept
+                [/has_attack]
+                is_enemy = yes
+            [/filter_adjacent]
+        [/filter]
+        [lua]
+            code=<<
+--attacker is e
+--defender is u
+local _ = wesnoth.textdomain "wesnoth-help"
+local c = wesnoth.current.event_context
+local x1 = c.x1
+local y1 = c.y1
+if x1 ~= c.unit_x or y1 ~= c.unit_y then
+    return
+end
+local u = wesnoth.units.get(x1,y1)
+
+local enemies = wesnoth.units.find_on_map({{"filter_adjacent",{id=u.id, is_enemy = true}},{"has_attack",{special_id = "intercept"}}})
+
+for iter,e in ipairs(enemies) do
+    local i = 1
+    local special
+    while e.attacks[i] ~= nil do
+        special = wml.find_child(e.attacks[i].specials,{id="intercept"})
+        if special ~= nil then
+            break
+        end
+        i = i + 1
+    end
+    if not e.variables.intercept_used then
+        e.variables.intercept_used = 0
+    end
+    if special.uses_per_turn and special.uses_per_turn <= e.variables.intercept_used then
+        table.remove(enemies,iter)
+    end
+end
+
+if #enemies == 0 then return end
+
+local animator =      wesnoth.units.create_animator()
+local sheath_weapon = wesnoth.units.create_animator()
+
+for iter,e in ipairs(enemies) do
+    e.variables.intercept_used = e.variables.intercept_used and e.variables.intercept_used + 1 or 1
```

Comment:
imo a ` e.variables.intercept_used = (e.variables.intercept_used or 0) + 1` is clearer